packages/core/src/utils/viaProxy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Utility to redirect remote sources via Proxy
+ *
+ * This utility is used to rewrite a request to be routed through
+ * the Proxy instead.
+ */
+
+/**
+ * Virtual Module provided by the Motion Canvas
+ * Plugin. Returns `false` if proxy is set to `false`, else `true`
+ */
+//@ts-ignore
+import {enabled} from 'virtual:proxysettings';
+
+/**
+ * Helper-Function to route remote requests through a local proxy.
+ *
+ * @example This rewrites a remote url like `https://via.placeholder.com/300.png/09f/fff` into a URI-Component-Encoded string like `/cors-proxy/https%3A%2F%2Fvia.placeholder.com%2F300.png%2F09f%2Ffff`
+ */
+export function viaProxy(url: string, overrideProxySetting?: boolean) {
+  const isEnabled = overrideProxySetting ?? enabled;
+  if (!isEnabled) {
+    // Proxy is disabled, so we just pass as-is.
+    return url;
+  }
+
+  if (url.startsWith('/cors-proxy/')) {
+    // Already proxied, return as-is
+    return url;
+  }
+
+  // window.location.hostname is being passed here to ensure that
+  // this does not throw an Error for same-origin requests
+  // e.g. /some/image -> localhost:9000/some/image
+
+  const selfUrl = new URL(window.location.toString());
+  // inside a try-catch in case the URL cannot be understood
+  try {
+    const expandedUrl = new URL(url, selfUrl);
+    if (!expandedUrl.protocol.startsWith('http')) {
+      // this is probably some embedded image (e.g. image/png;base64).
+      // don't touch and pass as is
+      return url;
+    }
+    if (selfUrl.host === expandedUrl.host) {
+      // This is a request to a "local" resource.
+      // No need to rewrite
+      return url;
+    }
+  } catch (_) {
+    // in case of error just silently pass as-is
+    return url;
```

Comment:
IMO a malformed or non-HTTP(S) URL should have an associated warning that the URL isn't being proxied, but maybe that's too verbose.

Reply:
i think that the Browser can do that for us — if it is indeed bad value.
Maybe there is some non-standard way of providing a resource that URL does not handle — in this case, the value should probably not be proxied anyways.

If the Browser cannot understand the passed value it will log to the console, if it can, it will interpret it correctly.